synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -231,10 +235,24 @@ def get_rules(self, context):
 
             # Loop through to see which member events we've seen and have rules
             # for and which we need to fetch
-            for key, event_id in current_state_ids.iteritems():
+            for key in current_state_ids:
                 if key[0] != EventTypes.Member:
                     continue
 
+                user_id = key[1]
```

Comment:
If we are optimising it might be worth noting that `a, b  = key` seems to be faster than `a = key[0], b = key[1]`